src/routers/xcm/Router.sol

```diff
@@ -44,6 +72,19 @@ contract ConnectorXCMRouter {
         _;
     }
 
+    //todo(nuno): add auth modifier
+    function file(bytes32 what, uint64 buyExecutionWeightLimit, uint64 transactWeightAtMost, uint256 feeAmount)
```

Comment:
@offerijns @AStox what auth modifier should we use here? `onlyConnector`, like we do for the `send` doesn't seem sensible here. Something like `auth` in the `Connector.sol`? I don't quite follow the `wards` idea tho? It is like a single-value whitelist of addresses allowed to make changes?

Reply:
Should be `auth`, and then we can add some multisig as a ward on the router.

The `wards` pattern is indeed just a `value whitelist of addresses allowed to make changes`. This can either be other smart contracts within the protocol, or other accounts such as a multisig.

Similarly, the goal with this pattern is to make it easy to audit what accounts have access on the contract.

There's a bit more context on this coding style here: https://github.com/makerdao/mips-best-practices#design-and-specification Generally for Tinlake we've followed all these guidelines except for the naming :)

Reply:
Also the versioning guideline is a bit out of date, this was a best practice in the past because the assumption was that old Solidity versions were more battle tested through time. But nowadays new releases are anyways tested more extensively and also MakerDAO is adopting >0.8 already.

Reply:
Thank you again, @offerijns 🚀  done 👍